styled-components required dep after upgrading to nx 9.2.2

## Expected Behavior

I can run the nx build command without npm installing `styled-components` package.

## Current Behavior

What is the current behavior?
After upgrading from nx 9.0.0 to nx 9.2.2, I can't build my react app.  I get the following msg

```
ERROR in ./app/App.jsx
Module not found: Error: Can't resolve 'styled-components' in 'foo/apps/bar/src/app'
```

We use `emotion` and never relied on `styled-components`. 

When I npm installed `styled-components`, then the build command works again.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

update to 9.2.2 and run `nx build <appname>`

@jaysoo - Thanks for looking into this.

Here's a repo: https://github.com/brianmcd/nx-emotion-bug.  I generated the project with the latest Nx and made the changes in [app.tsx](https://github.com/brianmcd/nx-emotion-bug/blob/master/apps/emotion-test/src/app/app.tsx).

I think using emotion's `css` prop instead of its styled API is triggering the behavior (https://emotion.sh/docs/css-prop#jsx-pragma).

Trying to build the project as-is results in: `Module not found: Error: Can't resolve 'styled-components'`. If you actually install `styled-components` in that repo, you'll see this in the bundled output, even though the code isn't using `styled-components`: 

```
/* harmony import */ var styled_components__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! styled-components */ "../../../node_modules/styled-components/dist/styled-components.browser.esm.js");
```

In my real project, I spliced out `babel-plugin-styled-components` and it fixed it for me:

```javascript
function removeStyledComponentsPlugin(nxConfig) {
  const options = getBabelLoaderOptions(nxConfig);
  const idx = options.plugins.findIndex((p) => {
    return Array.isArray(p) && p[0].includes('babel-plugin-styled-components');
  });
  if (idx === -1) {
    throw new Error();
  }
  options.plugins.splice(idx, 1);
}

function getBabelLoaderOptions(config) {
  const idx = config.module.rules.findIndex((r) => r.loader === 'babel-loader');
  if (idx === -1) {
    throw new Error();
  }
  return config.module.rules[idx].options;
}
```

--- a/packages/react/src/utils/babel-utils.ts
+++ b/packages/react/src/utils/babel-utils.ts
@@ -1,4 +1,9 @@
+import { appRootPath } from '@nrwl/workspace/src/utils/app-root';
+import { readJsonFile } from '@nrwl/workspace';
+import { join } from 'path';
+
 export function updateBabelOptions(options: any): void {
+  // Add react babel preset
   const idx = options.presets.findIndex(
     p => Array.isArray(p) && p[0].indexOf('@babel/preset-env') !== -1
   );
@@ -8,16 +13,27 @@ export function updateBabelOptions(options: any): void {
       useBuiltIns: true
     }
   ]);
-  options.plugins.splice(
-    0,
-    0,
-    [
+
+  // TODO: Remove this once we have composable webpack and babel plugins.
+
+  // Add babel plugin for styled-components or emotion.
+  // We don't have a good way to know when a project uses one or the other, so
+  // add the plugin only if the other style package isn't used.
+  const packageJson = readJsonFile(join(appRootPath, 'package.json'));
+  const deps = { ...packageJson.dependencies, ...packageJson.devDependencies };
+  const hasStyledComponents = !!deps['styled-components'];
+  const hasEmotion = !!deps['@emotion/core'];
+  if (hasStyledComponents && !hasEmotion) {
+    options.plugins.splice(0, 0, [
       require.resolve('babel-plugin-styled-components'),
       {
         pure: true
       }
-    ],
-    require.resolve('babel-plugin-emotion')
-  );
+    ]);
+  }
+  if (hasEmotion && !hasStyledComponents) {
+    options.plugins.splice(0, 0, require.resolve('babel-plugin-emotion'));
+  }
+
   return options;
 }

